galette/lib/Galette/Entity/Adherent.php

```diff
@@ -2414,4 +2416,89 @@ protected function getEventsPrefix(): string
     {
         return 'member';
     }
+
+
+    /**
+     * Get QR codes associated to member
+     *
+     * @return QrCode[]
+     */
+    public function getQrCodes(): array
+    {
+        global $routeparser, $login;
+
+        $qrcodes = [];
+
+        if (!$login->isAdmin() && !$login->isStaff() && !$login->isGroupManager()) {
+            //only admin, staff and group managers can get QR codes
+            return $qrcodes;
+        }
+
+        $qrcodes['vcard'] = new QrCode(
+            data: $this->getVCard()->serialize(),
+            label: $this->sname,
+            url: $routeparser->urlFor('memberVCard', ['id' => $this->id]),
+            logo: GALETTE_ROOT . '/includes/qr-logos/address-card-o.svg'
```

Comment:
Why not use the emojis included in Fomantic?

```suggestion
            logo: GALETTE_ROOT . '/webroot/themes/default/ui/themes/galette/assets/emojis/1faaa.svg'
```

Reply:
Just because I do not like the final render :/

Two main reasons: 
1- icons are full color while qr code is black/white (not very important though - and probably worse because of the second point), 
2- image is kind of "lost"  in the middle of the code; it's better with a white circle like I added on new made icons (those are FontAwasome ones modified to add the circle).

I guess I should address the second using a route that would modify the SVG source to add the circle (maybe not that easy, I've used Inkscape, not been able to find how to do that programmatically).

Reply:
It makes sense. I understand :+1:

Reply:
I just made a quick try with a colored icon, I'm not convinced ^^

<img width="353" height="352" alt="image" src="https://github.com/user-attachments/assets/2b96c5c7-c6eb-47d6-87c1-365eab3186aa" />

Reply:
Just keep your initial b&w version then :wink: